authbroker_client/views.py

```diff
@@ -2,22 +2,44 @@
 from django.shortcuts import redirect
 from django.http import HttpResponseBadRequest, HttpResponseServerError
 from django.conf import settings
-from django.contrib.auth import authenticate, login
+from django.contrib.auth import authenticate, login, REDIRECT_FIELD_NAME
+from django.utils.http import is_safe_url
+
+from authbroker_client.utils import (
+    get_client,
+    AUTHORISATION_URL,
+    TOKEN_URL,
+    TOKEN_SESSION_KEY,
+)
+
+
+REDIRECT_SESSION_FIELD_NAME = f"_oauth2_{REDIRECT_FIELD_NAME}"
+
 
-from authbroker_client.utils import get_client, AUTHORISATION_URL, TOKEN_URL, \
-    TOKEN_SESSION_KEY
 try:
     from raven.contrib.django.raven_compat.models import client
+
     capture_exception = client.captureException
 except ImportError:
     from sentry_sdk import capture_exception
 
 
+def get_next_url(request):
+    next_url = request.GET.get(
+        REDIRECT_FIELD_NAME,
+        request.session.get(REDIRECT_SESSION_FIELD_NAME)
+    )
+    if next_url and is_safe_url(next_url, allowed_hosts=settings.ALLOWED_HOSTS):
```

Comment:
Nice - `and is_safe_url()`

Reply:
saying that tho, do you maybe want to add a test to check that an 'unsafe' URL is _not_ returned as `next`?

Reply:
Always the key to testing ... put on evil hat and think of all the ways to trip up the code :+1: 

It looks like `is_safe_url` is getting deprecated in favour of `url_has_allowed_host_and_scheme`.

Reply:
Yep, I've added to a couple of tests to verify it only handles safe urls